services/datamanager/datasync/sync.go

```diff
@@ -124,69 +149,35 @@ func (s *syncer) SyncFile(path string, stopAfter time.Time) {
 	}
 	s.progressLock.Unlock()
 
-	for {
-		if s.cancelCtx.Err() != nil {
-			return
-		}
-
-		if time.Now().After(stopAfter) {
-			return
+	if !s.MarkInProgress(path) {
```

Comment:
I see this usage got moved, and now it's right after we acquire+release the `progressLock`.

This method also takes the `progressLock`. IIUC correctly, I think the [above check](https://github.com/viamrobotics/rdk/pull/4070/files#diff-3cb68c1958acbc6d7fe6d32c375af46ce12e78489d1e47aa4f69fa89228bcc07R145-R150) can just go away now?

Reply:
Makes sense to me!